### ISSUE TITLE
train accuracy keeps decreasing while train-loss also decrease?

### ISSUE THREAD

**Issue by zhaohan-xi** [Kaushalya/gnn-meta-attack-pytorch#4]
Hello,
This pytorch version of meta-gradient attack for original work is remarkable! But I have a slight question when running your code. 
During perturbation, I see the train-accuracy could obviously decrease, which means train-loss should increase, but the train-loss keeps decreasing at the same time. The more we trained, the more train-loss decrease. That is confusing, could I ask why this happen from this code? Thanks!

**Comment by Kaushalya** [comment]
Hi,
I assume you are refering to the loss printed in line [139 of meta_attack.py](https://github.com/Kaushalya/gnn-meta-attack-pytorch/blob/b1295aea0a605105838bbcbdce4402f55e76d319/meta_learn/meta_attack.py#L139). The training loss decreases since we update weights with respect to a pertuebed adjacency matrix. Can you provide an example?

For how many perturbations did you run? Accosrding to my observationns test accuracy increases if number of perturbations is less than 5% of edges.

**Comment by zhaohan-xi** [comment]
Hi,
Thanks for response! I see what may going wrong in training. In [line 47](https://github.com/Kaushalya/gnn-meta-attack-pytorch/blob/b1295aea0a605105838bbcbdce4402f55e76d319/meta_learn/meta_attack.py#L47), [line 151](https://github.com/Kaushalya/gnn-meta-attack-pytorch/blob/b1295aea0a605105838bbcbdce4402f55e76d319/meta_learn/meta_attack.py#L151) and [line 191](https://github.com/Kaushalya/gnn-meta-attack-pytorch/blob/b1295aea0a605105838bbcbdce4402f55e76d319/meta_learn/meta_attack.py#L191) of meta_attack.py, instead of feeding in "adj" or "adj_modified", should it first makes normalization of these adjacency matrices, no matter modified or not. The [preprocessing function](https://github.com/Kaushalya/gnn-meta-attack-pytorch/blob/b1295aea0a605105838bbcbdce4402f55e76d319/meta_learn/utils.py#L57) wrote in your utils.py is good to do that.
